Source/WebCore/platform/graphics/texmap/TextureMapperPlatformLayerBuffer.cpp

```diff
@@ -158,6 +158,11 @@ void TextureMapperPlatformLayerBuffer::paintToTextureMapper(TextureMapper& textu
         });
 }
 
+bool TextureMapperPlatformLayerBuffer::isHolePunchBuffer() const
+{
+    return m_extraFlags.contains(TextureMapperFlags::ShouldNotBlend);
+}
```

Comment:
I think we should add a comment here, at least for me it's not obvious why a buffer with the not blend flag is always a hole punch buffer.